### PR TITLE
Forbid `=` in group names

### PR DESCRIPTION
**Pull request by kobergj**
Since underlying `ldap` library panics when you do so, this is the best (aka quickest) way to solve this

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiGraph-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/20976/51/1) failed. Further test are cancelled...

**Review by rhafer — CHANGES_REQUESTED** [review]
I think this there is some problem with escaping here (which is likely not limited to the '=' charcter). It would be way better to correctly understand that problem before we merge this "fix".

(BTW, this might also be a problem with the libregraph/idm and it might just work when using a proper LDAP server)

**Comment by kobergj** [comment]
@rhafer I don't think it is related to escaping. When renaming a group we use the method [ModifyDN](https://pkg.go.dev/github.com/go-ldap/ldap#Conn.ModifyDN) which is expecting a string consisting of key-value pair separated by "=" 

Updating with group e.g. "grp=" leads to a sting like `cn=grp=`. This causes the `ldap` package to panic because it gets confused. 

I tried escaping "=" or quoting the names, both without success.

The best solution would probably be to fix the issue upstream. But I don't know how much effort that is

**Comment by rhafer** [comment]
Uff. You're right. `=` does not need to be escaped in LDAP DNs (https://www.rfc-editor.org/rfc/rfc4514)

There seems to be a bug in ldap.ParseDN() which causes it to choke on DN attribute value ending with `=`. (It will not return an error, but also not correctly parse it). And then idm panics on it because it does not expect an empty result when err is nil :boom: 

We should definetly create an upstream issue for this.

But if we want to disallow names with `=` for now, we should do that consistently through out the LDAP backend. (users, schools and classes have the same issue I think)

**Comment by kobergj** [comment]
@rhafer followed your suggestion. Please double check

File: services/graph/pkg/service/v0/service.go
Comment (rhafer) [inline services/graph/pkg/service/v0/service.go:432]: It would be nice if you'd mention the upstream issue here. I just found: https://github.com/go-ldap/ldap/issues/416 (even though it's talking about base64 encoded values, the issue is really about the `=` sign).

**Review by rhafer — APPROVED** [review]
@kobergj Look was just fixed a couple of days ago: https://github.com/go-ldap/ldap/pull/425 

Funny coincidence. I wonder if we should just wait for next go-ldap release. But I am also fine merging this. Would be nice if you'd mention the upstream issue in the code tough.

**Comment by kobergj** [comment]
Let's test with new version. If it works we can close this PR

**Comment by kobergj** [comment]
Superior fix here: https://github.com/owncloud/ocis/pull/6004

**Pull request closed without merging**
